libs/api/ebpf_api.cpp

```diff
@@ -3324,18 +3331,21 @@ ebpf_object_load(_Inout_ struct bpf_object* object) NO_EXCEPT_TRY
     }
 
     if (Platform::_is_native_program(object->file_name)) {
-        struct bpf_program* program = bpf_object__next_program(object, nullptr);
-        if (program == nullptr) {
-            EBPF_RETURN_RESULT(EBPF_INVALID_ARGUMENT);
+        result = _ebpf_program_load_native(object->file_name, object->execution_type, object);
+        if (result != EBPF_SUCCESS) {
+            EBPF_RETURN_RESULT(result);
+        }
+        for (auto& program : object->programs) {
+            if (program->autoload) {
+                continue;
+            }
+            if (program->fd > 0) {
+                Platform::_close(program->fd);
```

Comment:
It looks like this logic has been added at 2 places. Ideally it should be at only one place.

1. `ebpf_object_load()` -- this function
2. `_initialize_ebpf_programs_native()`, which is also called by `ebpf_object_load()`.

```
ebpf_object_load
	_ebpf_program_load_native
		_load_native_programs
		_initialize_ebpf_object_native
			_initialize_ebpf_programs_native
```

Reply:
Updated